server/server.go

```diff
@@ -1318,8 +1318,13 @@ func (s *Server) SetReplicationModeConfig(cfg config.ReplicationModeConfig) erro
 	return nil
 }
 
-// AddLeaderCallback adds a callback in the leader campaign phase.
-func (s *Server) AddLeaderCallback(callbacks ...func(context.Context)) {
+// IsPrimary returns if the server is leader.
+func (s *Server) IsPrimary() bool {
```

Comment:
ditto

Reply:
I will add more comments.

Reply:
how about change to more general names to **"IsServing()" and "AddServerReadyCallback"**? Preferred
Or change names to "IsPrimaryOrLeader" and "AddPrimaryOrLeaderCallback"?

Reply:
good idea